aries_cloudagent/admin/server.py

```diff
@@ -263,6 +265,27 @@ async def collect_stats(request, handler):
 
             middlewares.append(collect_stats)
 
+        @web.middleware
+        async def set_wallet(request, handler):
+            wallet_id = request.headers.get("x-wallet-id")
+            if wallet_id:
+                wallet_record = await WalletRecord.retrieve_by_id(
+                    self.context, wallet_id
+                )
+                # We amend the context settings so that the wallet that will be
+                # opened later has different context
+                self.context.settings = self.context.settings.extend(
```

Comment:
Modify the settings for the request context. This way any wallets that are injected after this point will use the modified wallet configuration and open different wallets.